exporter.go

```diff
@@ -113,13 +116,21 @@ func (e *Exporter) scrape(f func(prometheus.Collector)) {
 		e.scrapeHistogramMetric.Observe(time.Since(start).Seconds())
 	}()
 
-	// system stats
-	c, err := beanstalk.Dial("tcp", e.address)
+	// opens a tcp connection with connection timeout.
+	conn, err := net.DialTimeout("tcp", e.address, e.connectionTimeout)
 	if err != nil {
 		e.scrapeConnectionErrorMetric.Inc()
 		log.Fatalf("Error. Can't connect to beanstalk: %v", err)
 		return
 	}
+
```

Comment:
The defer has been moved to the line 134, but it should be right after this if statement otherwise the connection will leak if `SetReadDeadline` return an error.